### DISCUSSION TITLE
Install Winget-AutoUpdate via Powershell and GPO issue

### DISCUSSION THREAD

**Discussion by workpush** [Romanitho/Winget-AutoUpdate#550] [Q&A]
Hello everyone,

I want to use Winget-AutoUpdate keep the (winget) software within our domain on our desktop clients up-to-date in system and user context. The domain users do not have Admin rights.

1. For that I created a powershell script
```ps
#Install Winget-Autoupdate
$WAUParameters= "-Silent -InstallUserContext -DoNotUpdate -NotificationLevel Full -MaxLogFiles 4 -MaxLogSize 2097152 -DisableWAUAutoUpdate -DesktopShortcut -StartMenuShortcut -UpdatesAtLogon"
Start-Process "powershell.exe" -Argument "-NoProfile -ExecutionPolicy Bypass -Command ""& '$PSScriptRoot\Winget-AutoUpdate-Install.ps1' $WAUParameters""" -Wait -Verb RunAs
```
and saved that to `\domain.lan\SYSVOL\domain.lan\scripts\WAU\installWAU.ps1`

2. Download the latest release(1.19.1) from: https://github.com/Romanitho/Winget-AutoUpdate/releases
and extract the `Winget-AutoUpdate-Install.ps1'`, etc. directly in the directory `\domain.lan\SYSVOL\domain.lan\scripts\WAU\`

3. Create a GPO:
Computer Policies->Windows Settings->Scripts->Startup->Powershell and put `\domain.lan\SYSVOL\domain.lan\scripts\WAU\installWAU.ps1` there

So far, so good. It installs WAU and the tasks are registered:
`C:\ProgramData\Winget-AutoUpdate\logs\WAU-Installer.log` is saying:

```
################################################################
#     18.01.2024 -   INSTALLING WAU
################################################################
23:27:13 - Checking prerequisites...
23:27:14 - Prerequisites checked. OK

23:27:14 - Checking if WinGet is installed/up to date.
23:27:15 - -> WinGet is up to date: v1.6.3482

23:27:15 - Installing WAU...
23:27:20 - -> Updating previous installation...
23:27:20 - -> Copied brand new Blacklist.
23:27:26 - -> Installing WAU scheduled tasks
23:27:26 - -> Setting Registry config
23:27:26 - -> Checking Mods Directory:
23:27:26 -    The mods directory is secured!
23:27:26 - -> Checking Functions Directory:
23:27:27 -    The Functions directory is secured!
23:27:28 - -> WAU Installation succeeded!

23:27:29 - Skip running Winget-AutoUpdate
23:27:29 - End of process.
```

Whenever I log in as a domain user would except WAU to update the Software. I can see the task is executed in "Task Scheduler", WAU->Winget-AutoUpdate. But it is running forever and in `C:\ProgramData\Winget-AutoUpdate\logs\` no other log file is created. I also receive no notifications - even if I click the Desktop icon or the start menu. If I start the icons/application with right click->execute as admin does not make a difference.

If I run the `Gui.ps1` as DomainUser(rightclick->execute) or AdminUser(powershell as admin, navigate to `GUI.ps1` directory and execute `.\GUI.ps`") from the `WAU-Configurator`, both are showing: "WAU is currently installed v1.19.1.".

Any has a hint, what is going wrong here? Might this be a system/user context issue? 

Thank you very much!

PS: I also have the other issue to install software via winget-install: https://github.com/Romanitho/Winget-Install/discussions/102

**Comment by Romanitho** [comment]
check if everything is correctly copied to the destination folder on the computer (like functions and stuff)
